src/socket.cpp

```diff
@@ -142,6 +155,8 @@ void CSocket::Init ( const quint16 iPortNumber, const quint16 iQosNumber, const
 
         QObject::connect ( this, &CSocket::ServerFull, pServer, &CServer::OnServerFull );
     }
+
+    bIsInitRan = true; // QObject::connect once only
```

Comment:
If there are things that need to happen multiple times and other things that should only happen once - put them in separate methods.  Then you can decide which to call.  This method can then call the "multiple" times method - but only itself be called once.

Reply:
Thanks to you I've found a way to make it look a lot better now, no more `if ( bIsInitRan )`.